source/pidomus.cc

```diff
@@ -1200,11 +1200,20 @@ template <int dim, int spacedim, typename LAC>
 void
 piDoMUS<dim, spacedim, LAC>::set_constrained_dofs_to_zero(typename LAC::VectorType &v) const
 {
-  for (unsigned int i = 0; i < global_partitioning.n_elements(); ++i)
+  if (global_partitioning.n_elements() > 0)
     {
-      auto j = global_partitioning.nth_index_in_set(i);
-      if (constraints.is_constrained(j))
-        v[j] = 0;
+      auto k = global_partitioning.nth_index_in_set(0);
+      if (constraints.is_constrained(k))
+        v[k] = 0;
+      else
+        v[k] = v[k];
+      for (unsigned int i = 1; i < global_partitioning.n_elements(); ++i)
+        {
+          auto j = global_partitioning.nth_index_in_set(i);
+          if (constraints.is_constrained(j))
+            v[j] = 0;
+        }
+      v.compress(VectorOperation::insert);
```

Comment:
This is fine, and with this probably we shouldn't need the hack above. @nicola-giuliani , did you try without the hack above first, and it didn't work, or was the hack above necessary if one did not have the compress here?